### ISSUE TITLE
gapic: expose support for paginated List calls?

### ISSUE THREAD

**Issue by timburks** [googleapis/gapic-generator-go#473]
I would like to support pagination in a GraphQL wrapper for a gRPC API. If possible, I'd also like to use the Go GAPIC to implement this, but the generated List methods return an iterator that doesn't expose a cursor. Some possible options:

1. Modify the iterator to return a token for the iteration in progress (which seems hard).
2. Add another GAPIC method that has a paginated list interface (which seems ugly and potentially confusing).
3. Call the gRPC List methods directly (foregoing retry and any other advantages of GAPICs). For this it would be nice to have an accessor on the GAPIC connection that provides the underlying gRPC connection.

What do you think? As a workaround, I can just use gRPC directly with a separately-constructed non-GAPIC connection... and adding the following to my gapic package works well too:

```
package gapic

import rpcpb "github.com/apigee/registry/rpc"

func (c *RegistryClient) GrpcClient() rpcpb.RegistryClient {
	return c.registryClient
}
```

**Comment by timburks** [comment]
FWIW I'm using the following to add an accessor to the generated Go GAPIC that returns the underlying gRPC connection:
```
# add an accessor for the underlying gRPC client of the GAPIC client
cat >> gapic/registry_client.go <<END
func (c *RegistryClient) GrpcClient() rpcpb.RegistryClient {
	return c.registryClient
}
END
```

**Comment by noahdietz** [comment]
> Modify the iterator to return a token for the iteration in progress (which seems hard).

Well...it doesn't give you a cursor to the exact element it's on, but you can get the `page_token` if you want. Each generated Iterator has a [`PageInfo` accessor](https://github.com/googleapis/google-cloud-go/blob/master/asset/apiv1/asset_client.go#L723-L726), which gives you the [`iterator.PageInfo`](https://pkg.go.dev/google.golang.org/api/iterator#PageInfo) which has a `Token` field (`page_token`). The `pageInfo` is updated on a `FooIterator` instance when it makes another RPC to retrieve more items.

Does that work?

**Comment by timburks** [comment]
I think that might work - is this the page token for the current page of elements? (if so, then we would need to remember that plus the number of elements we've consumed) or is it the page token that we would use to get the next page of elements? (if so, it seems tricky to get back to any unused elements of the current page) 

For our purposes the accessor that I patched in is working fine. Thanks for thinking about this!
